Library/EnvelopeFactory.ts

```diff
@@ -153,15 +172,13 @@ class EnvelopeFactory {
 
     private static createDependencyData(telemetry: Contracts.DependencyTelemetry & Contracts.Identified): Contracts.Data<Contracts.RemoteDependencyData> {
         var remoteDependency = new Contracts.RemoteDependencyData();
-        if (typeof telemetry.name === "string") {
-            remoteDependency.name = telemetry.name.length > 1024 ? telemetry.name.slice(0, 1021) + "..." : telemetry.name;
-        }
-        remoteDependency.data = telemetry.data;
-        remoteDependency.target = telemetry.target;
+        remoteDependency.name = telemetry.name?.substring(0, MAX_NAME_LENGTH);
+        remoteDependency.data = telemetry.data?.substring(0, MAX_PROPERTY_LENGTH);
+        remoteDependency.target = telemetry.target?.substring(0, MAX_NAME_LENGTH);
```

Comment:
I believe the constants lose benefit when the name cannot be associated with all the things that will use it, like here, you use max name for target field, is way cleaner and easier to read if we have the actual number here instead of having constants, but is up to you I don't strongly oppose to current solution

Reply:
That makes sense. I wanted to avoid magic numbers, but since we're just following the specifications of another API I agree that just using the value could improve readability here.